### PR TITLE
UCT: added built-in device id 4120 to known device list

### PR DESCRIPTION
**Pull request by hoopoepg**
- this device is used on VM where SRIOV is configured

### CODE REVIEW

File: src/uct/ib/base/ib_device.c [hunk 1]
```diff
@@ -29,13 +29,14 @@ static ucs_stats_class_t uct_ib_device_stats_class = {
...
-  {0x02c9, 4119, "ConnectX-5",     UCT_IB_DEVICE_FLAG_MLX5_PRM, 38},
-  {0x02c9, 4121, "ConnectX-5",     UCT_IB_DEVICE_FLAG_MLX5_PRM, 40},
+  {0x02c9, 4099, "ConnectX-3",       UCT_IB_DEVICE_FLAG_MLX4_PRM, 10},
+  {0x02c9, 4103, "ConnectX-3 Pro",   UCT_IB_DEVICE_FLAG_MLX4_PRM, 11},
+  {0x02c9, 4113, "Connect-IB",       UCT_IB_DEVICE_FLAG_MLX5_PRM, 20},
+  {0x02c9, 4115, "ConnectX-4",       UCT_IB_DEVICE_FLAG_MLX5_PRM, 30},
+  {0x02c9, 4117, "ConnectX-4 LX",    UCT_IB_DEVICE_FLAG_MLX5_PRM, 28},
+  {0x02c9, 4119, "ConnectX-5",       UCT_IB_DEVICE_FLAG_MLX5_PRM, 38},
+  {0x02c9, 4121, "ConnectX-5",       UCT_IB_DEVICE_FLAG_MLX5_PRM, 40},
+  {0x02c9, 4120, "ConnectX-5 SRIOV", UCT_IB_DEVICE_FLAG_MLX5_PRM, 50},
```
Comment (alinask) [inline src/uct/ib/base/ib_device.c]: The official name is simply ConnectX-5 (without SRIOV). 
maybe it's priority should be lower than 4121's

Comment (hoopoepg) [reply src/uct/ib/base/ib_device.c]: ok, updated

**Comment by yosefe** [comment]
BTW we can add flags to the SRIOV device code, to make it work with GID by default (in another PR)

**Comment by hoopoepg** [comment]
is device id 4120 used in SRIOV only

**Comment by yosefe** [comment]
AFAIK yes, but we need to double check offline

**Comment by hoopoepg** [comment]
could be problems when we communicate native & SRIOV system.

**Comment by yosefe** [comment]
it's OK if native system will use local addressing, as long as the remote address will be global.
the address type should affect which address the iface advertises through uct_iface_get_address(). the remote address type (on the endpoint) should be determined by checking the packed address which was provided from remote side.

**Comment by hoopoepg** [comment]
Then we have to update address pack: for now it pack address values based on address type. It should be changed to pack complete address and make decision for address type on peer side

**Comment by yosefe** [comment]
currently we have address flags the the packed address (UCT_IB_ADDRESS_FLAG_xx) which provide this info

**Comment by hoopoepg** [comment]
Ok, I will look at this and propose solution tomorrow

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4619/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4620/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
bot:ornl:retest

**Pull request merged**